### PR TITLE
Add design document for PubSub

### PR DESCRIPTION
**Pull request by dc7303**
**What this PR does / why we need it**:
Considering how to implement the metadata update, we reviewed the PubSub design in detail. Share design documents created while analyzing for other developers.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation**:

```docs

```

**Checklist**:
- [x] Added relevant tests or not required
- [x] Didn't break anything

### CODE REVIEW

**Comment by dc7303** [comment]
First, I wrote the content I thought was necessary. Do you have any content that you would like to supplement?

**Comment by hackerwins** [comment]
@dc7303 Thank you for writing.
May I ask @mojosoeun to review this PR?

**Comment by dc7303** [comment]
@hackerwins Sure! :)
@mojosoeun Please honest feedback. :)

File: design/pub-sub.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+```
+
+And to manage the event delivery target, we are using the [PubSub pattern](https://en.wikipedia.org/wiki/Publish%E2%80%93subscribe_pattern). 
+We can learn more by looking at the [sync package](https://github.com/yorkie-team/yorkie/tree/main/yorkie/backend/sync) we are implementing.
+ 
+The order of operation is as follows.
+
+1. The WatchDocuments API creates a `Subscription` instance with the client and document keys.
+2. Subscription instance internally manages the `DocEvent channel`, and the WatchDocuments API uses a select statement to retrieve events passed to the Subscription instance.
+3. The client can deliver the event to the Subscription instances that are subscribing to the same document through the `PubSub.Publish(publisherID *time.ActorID, topic string, event DocEvent)` method.
```
Comment (mojosoeun) [inline design/pub-sub.md:35]: 3. The client can deliver the event to the Subscription instances that are subscribed to the same document through the `PubSub.Publish(publisherID *time.ActorID, topic string, event DocEvent) method.`

File: design/pub-sub.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+## Summary
+Yorkie needs to share events happening in documents between different clients. For this, we implement this feature using gRPC server-side stream and the PubSub pattern.
+
+### Goals
+We should be able to share events with other clients who are subscribing to Documents.
+
+## Proposal Details
+
+### How does it work?
+Yorkie implements WatchDocuments API using [gRPC server-side stream](https://grpc.io/docs/languages/go/basics/#server-side-streaming-rpc) to deliver the events that have occurred to other clients.
```
Comment (mojosoeun) [inline design/pub-sub.md:17]: Yorkie implements WatchDocuments API using [gRPC server-side streaming](https://grpc.io/docs/languages/go/basics/#server-side-streaming-rpc) to deliver the events that have occurred to other clients.

**Review by mojosoeun — COMMENTED** [review]
Thank you for your great design document! 👍 I left a few comments.

**Comment by dc7303** [comment]
@mojosoeun 
Thanks for the comment 👍 
I modified it, so please check it again :)

**Review by mojosoeun — APPROVED** [review]
LGTM!

**Pull request merged**